### PR TITLE
Filter static types from list of types known to serializer

### PR DESCRIPTION
**Pull request by ReubenBond**
See #5021

### CODE REVIEW

File: src/Orleans.CodeGenerator/CodeGenerator.cs [hunk 1]
```diff
@@ -488,11 +488,16 @@ private bool ValidForKnownTypes(INamedTypeSymbol type)
                 case TypeKind.TypeParameter:
                 case TypeKind.Submission:
                 {
-                    if (this.log.IsEnabled(LogLevel.Trace)) this.log.LogTrace($"{nameof(ValidForKnownTypes)} skipping type {type} of kind {type.TypeKind}");
+                    // if (this.log.IsEnabled(LogLevel.Trace)) this.log.LogTrace($"{nameof(ValidForKnownTypes)} skipping type {type} of kind {type.TypeKind}");
```
Comment (sergeybykov) [inline src/Orleans.CodeGenerator/CodeGenerator.cs]: Why comment this line out?

Comment (ReubenBond) [reply src/Orleans.CodeGenerator/CodeGenerator.cs]: To match the other commented out lines, no real reason. I could delete them all or guard them using a different condition

Comment (sergeybykov) [reply src/Orleans.CodeGenerator/CodeGenerator.cs]: I think we either leave them uncommented and guarded by 
```csharp
if (this.log.IsEnabled(LogLevel.Trace))
```
or delete them if they aren't needed. We try not to leave commented out code in the codebase as a general rule.

Comment (ReubenBond) [reply src/Orleans.CodeGenerator/CodeGenerator.cs]: deleted

**Pull request merged**